### ISSUE TITLE
Some tests timeout intermittently when running in CI

### ISSUE THREAD

**Issue by brett19** [couchbase/gocbcorex#106]
We saw this during a CI run today:
```
2023-04-25T16:55:13.636Z	INFO	client	gocbcorex/kvclient.go:324	orphaned response encountered	{"poolId": "e015076e", "clientId": "e09be356", "opaque": "4", "opcode": "SASLAuth"}
panic: test timed out after 10m0s
```
when running the test `TestKvClientPoolClose`

**Comment by brett19** [comment]
Another case of this happening:
```
=== RUN   TestClientManagerClose
2023-04-25T19:19:45.727Z	DEBUG	gocbcorex/kvclientmanager.go:117	reconfiguring	{"mgrId": "e0ae62df"}
2023-04-25T19:19:45.727Z	DEBUG	pool	gocbcorex/kvclientpool.go:106	id assigned for 127.0.0.1:11210	{"mgrId": "e0ae62df", "poolId": "20992759"}
2023-04-25T19:19:45.727Z	DEBUG	pool	gocbcorex/kvclientpool.go:119	checking connections	{"mgrId": "e0ae62df", "poolId": "20992759"}
2023-04-25T19:19:45.727Z	DEBUG	pool	gocbcorex/kvclientpool.go:157	needs new clients	{"mgrId": "e0ae62df", "poolId": "20992759", "needed": 3}
2023-04-25T19:19:45.727Z	DEBUG	pool	gocbcorex/kvclientpool.go:426	no client found in fast map	{"mgrId": "e0ae62df", "poolId": "20992759"}
2023-04-25T19:19:45.727Z	DEBUG	pool	gocbcorex/kvclientpool.go:199	creating new client	{"mgrId": "e0ae62df", "poolId": "20992759"}
2023-04-25T19:19:45.727Z	DEBUG	pool	gocbcorex/kvclientpool.go:199	creating new client	{"mgrId": "e0ae62df", "poolId": "20992759"}
2023-04-25T19:19:45.727Z	DEBUG	pool.client	gocbcorex/kvclient.go:127	id assigned for 127.0.0.1:11210	{"mgrId": "e0ae62df", "poolId": "20992759", "clientId": "5a7cd591"}
2023-04-25T19:19:45.727Z	DEBUG	pool.client	gocbcorex/kvclient.go:127	id assigned for 127.0.0.1:11210	{"mgrId": "e0ae62df", "poolId": "20992759", "clientId": "e7104813"}
2023-04-25T19:19:45.727Z	DEBUG	pool	gocbcorex/kvclientpool.go:199	creating new client	{"mgrId": "e0ae62df", "poolId": "20992759"}
2023-04-25T19:19:45.727Z	DEBUG	pool.client	gocbcorex/kvclient.go:127	id assigned for 127.0.0.1:11210	{"mgrId": "e0ae62df", "poolId": "20992759", "clientId": "e3e604e6"}
2023-04-25T19:19:45.728Z	DEBUG	pool.client	gocbcorex/kvclient.go:211	bootstrapping	{"mgrId": "e0ae62df", "poolId": "20992759", "clientId": "e3e604e6"}
2023-04-25T19:19:45.732Z	DEBUG	pool.client	gocbcorex/kvclient.go:211	bootstrapping	{"mgrId": "e0ae62df", "poolId": "20992759", "clientId": "5a7cd591"}
2023-04-25T19:19:45.733Z	DEBUG	pool.client	gocbcorex/kvclient.go:211	bootstrapping	{"mgrId": "e0ae62df", "poolId": "20992759", "clientId": "e7104813"}
2023-04-25T19:19:45.739Z	DEBUG	pool.client	gocbcorex/kvclient.go:227	successfully bootstrapped new KvClient	{"mgrId": "e0ae62df", "poolId": "20992759", "clientId": "e3e604e6", "features": [4,6,7,8,10,11,14,15,16,17,18,20,23,25]}
2023-04-25T19:19:45.739Z	DEBUG	pool	gocbcorex/kvclientpool.go:119	checking connections	{"mgrId": "e0ae62df", "poolId": "20992759"}
2023-04-25T19:19:45.739Z	INFO	gocbcorex/kvclientmanager.go:265	closing kv client manager	{"mgrId": "e0ae62df"}
2023-04-25T19:19:45.739Z	INFO	pool	gocbcorex/kvclientpool.go:493	closing	{"mgrId": "e0ae62df", "poolId": "20992759"}
2023-04-25T19:19:45.739Z	INFO	pool.client	gocbcorex/kvclient.go:302	closing	{"mgrId": "e0ae62df", "poolId": "20992759", "clientId": "e3e604e6"}
2023-04-25T19:19:45.739Z	DEBUG	pool.client	memdx/client.go:110	cancelling operation	{"mgrId": "e0ae62df", "poolId": "20992759", "clientId": "5a7cd591", "opaque": 4}
2023-04-25T19:19:45.739Z	DEBUG	pool.client	gocbcorex/kvclient.go:220	bootstrap failed	{"mgrId": "e0ae62df", "poolId": "20992759", "clientId": "5a7cd591", "error": "request cancelled: context canceled"}
2023-04-25T19:19:45.739Z	INFO	pool.client	gocbcorex/kvclient.go:302	closing	{"mgrId": "e0ae62df", "poolId": "20992759", "clientId": "5a7cd591"}
2023-04-25T19:19:45.739Z	DEBUG	pool	gocbcorex/kvclientpool.go:207	closed during new client creation	{"mgrId": "e0ae62df", "poolId": "20992759"}
2023-04-25T19:19:45.741Z	DEBUG	pool.client	memdx/client.go:110	cancelling operation	{"mgrId": "e0ae62df", "poolId": "20992759", "clientId": "e7104813", "opaque": 6}
net.(*conn).Read(0xc0000aa250, {0xc0000ee708?, 0xf?, 0x0?})
	/opt/hostedtoolcache/go/1.19.8/x64/src/net/net.go:183 +0x45
io.ReadAtLeast({0x8c5540, 0xc0000aa250}, {0xc0000ee708, 0x18, 0x18}, 0x18)
	/opt/hostedtoolcache/go/1.19.8/x64/src/io/io.go:332 +0x9a
io.ReadFull(...)
	/opt/hostedtoolcache/go/1.19.8/x64/src/io/io.go:351
github.com/couchbase/gocbcorex/memdx.(*PacketReader).ReadPacket(0xc00037bc40?, {0x8c5540, 0xc0000aa250}, 0xc00044d580)
	/home/runner/work/gocbcorex/gocbcorex/memdx/packetreader.go:22 +0xcf
github.com/couchbase/gocbcorex/memdx.(*Conn).ReadPacket(...)
	/home/runner/work/gocbcorex/gocbcorex/memdx/conn.go:58
github.com/couchbase/gocbcorex/memdx.(*Client).run(0xc00037bc40)
	/home/runner/work/gocbcorex/gocbcorex/memdx/client.go:64 +0x85
created by github.com/couchbase/gocbcorex/memdx.NewClient
	/home/runner/work/gocbcorex/gocbcorex/memdx/client.go:55 +0x265

goroutine 274 [chan receive, 9 minutes]:
github.com/couchbase/gocbcorex.kvClient_SimpleCall[...]({0x8c8ab8, 0xc00037b880}, 0xc00043f0e0, {{0x8c9d50, 0xb231f8?}}, 0x85b8a0?, 0x82)
	/home/runner/work/gocbcorex/gocbcorex/kvclient_ops.go:91 +0x4b4
github.com/couchbase/gocbcorex.(*kvClient).bootstrap(...)
	/home/runner/work/gocbcorex/gocbcorex/kvclient_ops.go:132
github.com/couchbase/gocbcorex.NewKvClient({0x8c8ab8, 0xc00037b880}, 0xc0000af630, 0xc0002a3d58)
	/home/runner/work/gocbcorex/gocbcorex/kvclient.go:212 +0x9ba
github.com/couchbase/gocbcorex.NewKvClientPool.func1({0x8c8ab8, 0xc00037b880}, 0xd0?)
	/home/runner/work/gocbcorex/gocbcorex/kvclientpool.go:98 +0xb9
github.com/couchbase/gocbcorex.(*kvClientPool).startNewClientLocked.func1()
	/home/runner/work/gocbcorex/gocbcorex/kvclientpool.go:200 +0xd5
created by github.com/couchbase/gocbcorex.(*kvClientPool).startNewClientLocked
	/home/runner/work/gocbcorex/gocbcorex/kvclientpool.go:198 +0x258
```

**Comment by chvck** [comment]
The orphan response bit is a red herring here I believe. The issue here is that bootstrap is being cancelled but somehow kvclient simple op handler is never being called, leading to hanging on the channel.
